test/e2e/case46_dryrun_cli_test.go

```diff
@@ -0,0 +1,333 @@
+// Copyright (c) 2025 Red Hat, Inc.
+// Copyright Contributors to the Open Cluster Management project
+
+package e2e
+
+import (
+	"bytes"
+	"fmt"
+	"os"
+	"path/filepath"
+	"slices"
+	"strings"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"gopkg.in/yaml.v3"
+
+	"open-cluster-management.io/config-policy-controller/pkg/dryrun"
+	"open-cluster-management.io/config-policy-controller/test/utils"
+)
+
+const (
+	case46TestDataPath           = "../dryrun/"
+	policyYAML                   = "policy.yaml"
+	resourceYamlPrefix           = "input"
+	namespaceYamlPrefix          = "input_ns"
+	errOutputFile                = "error.txt"
+	defaultOutputFile            = "output.txt"
+	outputFromClusterFile        = "output_from_cluster.txt"
+	desiredStatusFile            = "desired_status.yaml"
+	desiredStatusFromClusterFile = "desired_status_from_cluster.yaml"
+)
+
+// persistentNamespaces are namespaces that should not be deleted after this test case
+// because they persist across test suites or are system namespaces
+var persistentNamespaces = []string{
+	"managed",
+	"default",
+}
+
+var untrackedMetadata = []string{
+	"resourceVersion:",
+	"generatedName:",
+	"creationTimestamp:",
+	"deletionTimestamp:",
+	"selfLink:",
+	"uid:",
+	"metadata.kubernetes.io/metadata.name:",
+	"kubernetes.io/metadata.name:",
+}
+
+var _ = Describe("Testing dryrun CLI", Ordered, func() {
+	var testNamespaces []string
+
+	BeforeAll(func() {
+		// Setup test namespaces before creating any resources
+		testNamespaces, err := findInputNamespaces(case46TestDataPath)
+		Expect(err).ToNot(HaveOccurred())
+
+		for _, nsFilePath := range testNamespaces {
+			utils.Kubectl("apply", "-f", nsFilePath)
+		}
+	})
+
+	AfterAll(func() {
+		for _, nsFilePath := range testNamespaces {
+			if shouldDelete, err := containsManagedNamespace(nsFilePath); shouldDelete && err == nil {
+				utils.KubectlDelete("-f", nsFilePath, "--wait")
+			}
+		}
+	})
+
+	testScenarios := []struct {
```

Comment:
I wanted to document which test folders I reused from `test/dryrun/`. Let me know if you have suggestions for a cleaner pattern to follow, especially when the test folders are nested e.g. `no_name` and `no_name/with_object_selector`